src/base/ArrayUtils.hh

```diff
@@ -56,6 +60,237 @@ template<class T, size_type N>
 inline CELER_FUNCTION bool is_soft_unit_vector(const Array<T, N>& v);
 
 //---------------------------------------------------------------------------//
-} // namespace celeritas
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+namespace detail
+{
+//---------------------------------------------------------------------------//
+//! Traits for operations on Real3 vectors
+template<class T>
+struct RealVecTraits;
+
+template<>
+struct RealVecTraits<float>
+{
+    //! Threshold for rotation
+    static CELER_CONSTEXPR_FUNCTION float min_accurate_sintheta()
+    {
+        return 0.07f;
+    }
+};
+
+template<>
+struct RealVecTraits<double>
+{
+    //! Threshold for rotation
+    static CELER_CONSTEXPR_FUNCTION double min_accurate_sintheta()
+    {
+        return 0.005;
+    }
+};
+
+} // namespace detail
```

Comment:
I completely subverted the rules when I put this `detail` block in a `.i.hh` file -- could you please correct my error and put this in a `detail/ArrayUtilsImpl.hh` file?